pages/api/v1/players/index.ts

```diff
@@ -0,0 +1,224 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import SQL from 'sql-template-strings';
+import Cors from 'cors';
+import { query } from '../../../../lib/db';
+import use from '../../../../lib/middleware';
+
+const cors = Cors({
+  methods: ['GET', 'HEAD'],
+});
+
+interface MasterPlayer {
+  PlayerID: number;
+  TeamID: number;
+  FranchiseID: number;
+  LeagueID: number;
+  SeasonID: number;
+  'First Name': string;
+  'Last Name': string;
+  'Nick Name': string;
+  Height: string;
+  Weight: string;
+  DOB: string;
+  Birthcity: string;
+  Birthstate: string;
+  Nationality_One: string;
+  Nationality_Two: string;
+  Nationality_Three: string;
+  position: string;
+}
+
+const getBasePlayerData = async (league, season) => await query(SQL`
+  SELECT *
+  FROM player_ratings
+  WHERE LeagueID=${+league}
+    AND SeasonID=${season.SeasonID}
+  INNER JOIN player_master
+  ON player_ratings.PlayerID = player_master.PlayerID
+  AND player_ratings.SeasonID = player_master.SeasonID 
+  AND player_ratings.LeagueID = player_master.LeagueID;
+`);
+
+const getSkaterStats = async (league, season) => await query(SQL`
+  SELECT *
+  FROM player_skater_stats_rs
+  WHERE LeagueID=${+league}
+    AND SeasonID=${season.SeasonID};
+`);
+
+const getGoalieStats = async (league, season) => await query(SQL`
+  SELECT *
+  FROM player_goalie_stats_rs
+  WHERE LeagueID=${+league}
+    AND SeasonID=${season.SeasonID};
+`);
+
+const getPlayerInfo = (player: MasterPlayer) => ({
+  id: player.PlayerID,
+  league: player.LeagueID,
+  season: player.SeasonID,
+  name: player['Last Name'],
+  team: player.TeamID,
+  position: player.position
+});
+
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  await use(req, res, cors);
+
+  const { league = 0, season: seasonid } = req.query;
+
+  const [season] =
+    (!Number.isNaN(+seasonid) && [{ SeasonID: +seasonid }]) ||
+    (await query(SQL`
+      SELECT DISTINCT SeasonID
+      FROM player_master
+      WHERE LeagueID=${+league}
+      ORDER BY SeasonID DESC
+      LIMIT 1
+  `));
+
+  const basePlayerData = await getBasePlayerData(league, season);
+  const skaterStats = await getSkaterStats(league, season);
+  const goalieStats = await getGoalieStats(league, season);
```

Comment:
We are running 3 different queries right here synchronously. This seems like it may take a while. Can we use a Promise.all to run them concurrently. For instance

`const [basePlayerData, skaterStats, goalieStats] = await Promise.all([getBasePlayerData(league, season), getSkaterStats(league, season), getGoalieStats(league, season)]);`

or something like this but a bit more readable?